### PR TITLE
MM-54201 Refactor mention parsing in preparation for multi-word mentions

### PR DESCRIPTION
**Pull request by hmhealey**
#### Summary
This is the PR to test these changes and merge this short-lived feature branch back into master. The changes are covered by these 3 PRs which have been reviewed individually:
1. https://github.com/mattermost/mattermost/pull/24932
2. https://github.com/mattermost/mattermost/pull/24936
3. https://github.com/mattermost/mattermost/pull/24937 (the one that is meaningfully changing things)

Dev review on this can be lighter since these changes were individually reviewed, but this is where the actual testing gets to happen

#### Ticket Link
MM-54201

#### Release Note
```release-note
Performed cleanup in preparation for adding multi-word mention support
```

### CODE REVIEW

**Review by larkox — APPROVED** [review]

**Review by yasserfaraazkhan — APPROVED** [review]
thank you Harrison. I tested this on browser Mac , window desktop app and mobile app.

**Review by lieut-data — APPROVED** [review]
Approving as a feature merge from the previous reviews without a detailed second review -- I did look at each of the other reviews previously.

**Pull request merged**
